cli.js

```diff
@@ -158,7 +158,12 @@ function exit(results) {
 	// correctly flush the output when multiple test files
 	process.stdout.write('');
 
-	process.exit(failed > 0 ? 1 : 0);
+	// Individually kill each child process.
+	Promise.map(results, function (result) {
+		return result.kill();
```

Comment:
Specifically, 
`nyc` uses `signal-exit`, which intercepts the kill signal and allows it to finish up writing files to disk before the process is actually killed.

I updated [jamestalmage/process-kill-is-sync](https://github.com/jamestalmage/process-kill-is-sync) to illustrate `on-exit` delaying the exit.